src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -2581,6 +2581,40 @@ module Crystal
       false
     end
 
+    def visit(node : OffsetOf)
+      @in_type_args += 1
+      node.offsetof_type.accept self
+      @in_type_args -= 1
+
+      type = node.offsetof_type.type?
+
+      node.offsetof_type.raise "type #{type} can't have instance variables" unless type.is_a?(InstanceVarContainer)
+      node.offsetof_type.raise "can't use typeof inside offsetof expression" if node.offsetof_type.is_a?(TypeOf)
+
+      instance_vars = type.all_instance_vars
+      ivar_name = node.instance_var.as(InstanceVar).name
+      ivar_index = instance_vars.keys.index(ivar_name)
```

Comment:
All of this can be replaced with `type.index_of_instance_var(node.instance_var.as(InstanceVar).name)` (extracting `node.instance_var.as(InstanceVar).name` to `ivar_name` is fine)